status-im/status-react
# Push notifications support #4432

### Summary:

Basic support of desktop notifications

### Review notes:

- react-native-desktop-notification native module introduced
- Fork of SnoreNotify library ( https://github.com/status-im/snorenotify.git ) used as backend to cover all notifications related functionality
- appimagetool is used to produce AppImage on Jenkins


### Testing notes (optional):
- Current PR introduces receiving of notifications for any incoming message even if app window is opened

### Steps to test:
- Open Status on primary testing machine
- Navigate to some chat
- Open another Status instance on another device
- Send the message into chat opened on primary testing machine
- Desktop notification "New Message" with message content should be triggered

status: ready

Usage of snorenotify introduced dependency on additional dev package to be installed in development environment:

Linux:
sudo apt install extra-cmake-modules

MacOS:
brew install kde-mac/kde/kf5-extra-cmake-modules


<blockquote><img src="https://avatars3.githubusercontent.com/u/11767950?s=400&v=4" width="48" align="right"><div><img src="https://assets-cdn.github.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/status-im/snorenotify">status-im/snorenotify</a></strong></div><div>snorenotify - Snorenotify is a multi platform Qt notification framework. Using a plugin system it is possible to create notifications with many different notification systems on Windows, Mac OS and...</div></blockquote>
